zaza/openstack/charm_tests/heat/setup.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python3
+#
+# Copyright 2019 Canonical Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import tenacity
+
+import zaza.model
+
+@tenacity.retry(stop=tenacity.stop_after_attempt(3),
+                wait=tenacity.wait_exponential(
+                multiplier=1, min=2, max=10))
+def domain_setup(application_name='heat'):
+    """Run required action for a working Heat unit."""
+    # Action is REQUIRED to run for a functioning heat deployment
+    logging.info('Running domain-setup action on heat unit...')
+    unit = zaza.model.get_units(application_name)[0]
+    zaza.model.block_until_unit_wl_status(unit.entity_id, "active")
+    zaza.model.run_action(unit.entity_id, "domain-setup")
+    zaza.model.block_until_unit_wl_status(unit.entity_id, "active")
```

Comment:
Add newline at the end of the file, and use `block_until_wl_status_info_starts_with` as above.